src/kernelspace/exec_monitor.c

```diff
@@ -25,14 +25,42 @@ struct {
 } ringbuf SEC(".maps");
 
 long ringbuffer_flags = 0;
+int max_n_proc;
+int n_monitored_proc;
+
+int filter_by_ppid;
+int ppid_list[10];
+int ppid_list_size;
 
 SEC("lsm/bprm_committed_creds")
 void BPF_PROG(exec_audit, struct linux_binprm *bprm)
 {
+	if (max_n_proc != 0 && n_monitored_proc == max_n_proc)
+		return;
+
 	long pid_tgid;
+	int ppid, selected_ppid = 0;
 	struct process_info *process;
 	struct task_struct *current_task;
 
+	// Get the parent pid
+	current_task = (struct task_struct *)bpf_get_current_task();
+	ppid = (int)BPF_CORE_READ(current_task, real_parent, pid);
+
+	if (filter_by_ppid) {
+		for (int i = 0; i < 10; i++) {
```

Comment:
`i < ppid_list_size` should be enough. If you don't trust ppid_list_size you could also do `i < ppid_list_size && i < PPID_LIST_MAP_LEN`

Reply:
Note that you can just drop filter_by_ppid if your loop is bounded by < ppid_list_size already

Reply:
So isn't the upper limit supposed to be a constant value? I am asking because, if I write i < ppid_list_size, then I get an error like this: invalid access to map value, value_size=60 off=60 size=4
R2 min value is outside of the allowed memory range
processed 184 insns (limit 1000000) max_states_per_insn 1 total_states 13 peak_states 13 mark_read 2

and, of course, fails to load the object.

It works with i < 10, this is why my code looks like this.

Reply:
Indeed, the BPF verifier wants to see a constant upper bound.
I suggested `i < ppid_list_size && i < PPID_LIST_MAP_LEN` which has both a constant upper bound and a variable bound, this should work. ;)

Reply:
It does not work, I tested it. It shows the same error, and I suspect it's from this: i < ppid_list_size && i < PPID_LIST_MAP_LEN

Reply:
Mhh ok this is rather weird since this should translate to the same bytecode but I'll trust you on that then.